evalml/tests/data_checks_tests/test_invalid_targets_data_check.py

```diff
@@ -385,4 +411,27 @@ def test_invalid_target_data_check_multiclass_problem_almostcontinuous_data():
                                                                              "errors": []}
 
     y_multiclass_low_classes = pd.Series(list(range(0, 3)) * 100)  # 2 classes, 300 samples, .01 class/sample ratio
+    X = pd.DataFrame({"col": range(len(y_multiclass_low_classes))})
     assert invalid_targets_check.validate(X, y=y_multiclass_low_classes) == {"warnings": [], "errors": []}
+
+
+def test_invalid_target_data_check_mismatched_indices():
```

Comment:
I think these are really good tests and well written.  I like that I can read them and understand exactly what the check is validating for.  I think this is fine as is, but since I don't understand entirely how this propagates to the user...I'll ask : is it useful ultimately for there to be a distinction between the length of the index being the same and the values being different and vice versa?  If not, then let's continue mission.  If so, an issue to further augment works.